geode-core/src/test/java/org/apache/geode/management/internal/ManagementAgentTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.internal;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.management.MBeanServer;
+import javax.management.remote.JMXConnectorServer;
+import javax.management.remote.JMXServiceURL;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.security.SecurityService;
+
+public class ManagementAgentTest {
+
+  // private ManagementAgent managementAgent = spy(ManagementAgent.class);
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Test
+  public void testSetMBeanServer() throws IOException {
+    DistributionConfig distributionConfig = mock(DistributionConfig.class);
+    InternalCache internalCache = mock(InternalCache.class);
+    JmxRmiSerialFilter serialFilter = mock(JmxRmiSerialFilter.class);
+    SecurityService securityService = mock(SecurityService.class);
+    when(internalCache.getSecurityService()).thenReturn(securityService);
+    when(securityService.isIntegratedSecurity()).thenReturn(false);
+    File tempFile = temporaryFolder.newFile("testFile");
+    when(distributionConfig.getJmxManagerAccessFile()).thenReturn(tempFile.getCanonicalPath());
+    ManagementAgent managementAgent =
+        new ManagementAgent(distributionConfig, internalCache, serialFilter);
+    MBeanServer mBeanServer = mock(MBeanServer.class);
+    JMXConnectorServer jmxConnectorServerWithMBeanServer = new JMXConnectorServer(mBeanServer) {
```

Comment:
I think you should move this to a static inner-class or better yet use Mockito to mock it:
```
JMXConnectorServer jmxConnectorServerWithMBeanServer = mock(JMXConnectorServer.class);
```

Reply:
I need two instances of subclass of `JMXConnectorServer`, which an abstract class. One instance with a non-null `mbeanServer` field, the other with a null `mbeanServer`. I don't know why move this to a static inner-class. I have tried to mock `JMXConnectorServer`. But I don't think a mock will work in this test case. I need to set the `mbeanServer` field of `JMXConnectorServer`. There is no setter for it. Only the constructor sets it.